ui/Modal/Org/ConfigureEns/UpdateMetadata.svelte

```diff
@@ -43,54 +44,71 @@
 
   async function setRecords() {
     buttonsDisabled = true;
-    submitButtonCopy = "Waiting for transaction confirmation...";
-
-    try {
-      let records: {
-        name: keyof ensResolver.Registration;
-        value: string | undefined;
-      }[] = [
-        { name: "address", value: orgAddress },
-        { name: "url", value: urlValue },
-        { name: "avatar", value: avatarValue },
-        { name: "twitter", value: twitterValue },
-        { name: "github", value: githubValue },
-      ];
-
-      // Filter out unchanged records.
-      records = records.filter(r => {
-        const existingValue = registration[r.name];
-
-        const normalizedExistingValue =
-          typeof existingValue === "string"
-            ? existingValue.toLowerCase()
-            : existingValue;
-
-        if (
-          r.value === undefined ||
-          (normalizedExistingValue === null && r.value === "")
-        ) {
-          false;
-        } else {
-          return normalizedExistingValue !== r.value.toLowerCase();
-        }
+
+    let records: {
+      name: keyof ensResolver.Registration;
+      value: string | undefined;
+    }[] = [
+      { name: "address", value: orgAddress },
+      { name: "url", value: urlValue },
+      { name: "avatar", value: avatarValue },
+      { name: "twitter", value: twitterValue },
+      { name: "github", value: githubValue },
+    ];
+
+    // Filter out unchanged records.
+    records = records.filter(r => {
+      const existingValue = registration[r.name];
+
+      const normalizedExistingValue =
+        typeof existingValue === "string"
+          ? existingValue.toLowerCase()
+          : existingValue;
+
+      if (
+        r.value === undefined ||
+        (normalizedExistingValue === null && r.value === "")
+      ) {
+        false;
+      } else {
+        return normalizedExistingValue !== r.value.toLowerCase();
+      }
+    });
+
+    if (records.length > 0) {
+      const updateNotification = notification.info({
+        message:
+          "Waiting for you to confirm the metadata update transaction in your connected wallet",
+        showIcon: true,
+        persist: true,
       });
 
-      if (records.length > 0) {
-        await ensResolver.setRecords(
+      let tx: transaction.ContractTransaction | undefined = undefined;
+      try {
+        tx = await ensResolver.setRecords(
           registration.domain,
           records as ensResolver.EnsRecord[]
         );
+        transaction.add(transaction.updateEnsMetadata(tx));
+        updated = true;
+      } catch (err) {
+        buttonsDisabled = false;
+        error.show(
+          new error.Error({
+            message: err.message,
+            source: err,
+          })
+        );
+      } finally {
+        updateNotification.remove();
       }
 
+      if (tx) {
```

Comment:
Instead of this `if`, could we have an early `return` in the `catch` block? Then typescript will know that `if` is defined.

Reply:
88a6ff9fc0396effafc654ed334f10895cf2c69d